internal/controller/certificates/policies/checks.go

```diff
@@ -112,10 +112,11 @@ func SecretKeystoreFormatMatchesSpec(input Input) (string, string, bool) {
 
 	if input.Certificate.Spec.Keystores.JKS != nil {
 		if input.Certificate.Spec.Keystores.JKS.Create {
-			if len(input.Secret.Data[cmapi.JKSSecretKey]) == 0 ||
-				len(input.Secret.Data[cmapi.JKSTruststoreKey]) == 0 {
-				return SecretMismatch, "JKS Keystore keys does not contain data", true
+			if len(input.Secret.Data[cmapi.JKSSecretKey]) == 0 {
+				return SecretMismatch, "JKS Keystore key does not contain data", true
 			}
+			// Can't check the Secrets truststore.jks (cmapi.JKSTruststoreKey) as
```

Comment:
at the check, we only have the cmapi.Certificate data and the corev1.Secret. So IMHO I can only check like: 

```go
if input.Secret.Data[cmapi.PKCS12TruststoreKey] != nil &&
    len(input.Secret.Data[cmapi.PKCS12TruststoreKey]) == 0 {
    return SecretMismatch, "PKCS12 Truststore key does not contain data", true
}
```

This will only check if when it already exists, then it should not be empty. IMHO this has not much value? But I still don't know if the CA needs to exist or not.  There is no check like `input.Certificate.Spec.Keystores.JKS.Create` to know whether the CA needs to exist or not.

Reply:
> This will only check if when it already exists, then it should not be empty. IMHO this has not much value

I think it has the same value as all the other checks that ensure that if, for example, the secret was manually modified in the cluster, those changes will be overriden.

Reply:
> the corev1.Secret

The Secret has CA if the issuer returns a CA cert and does not have it if it does not, so you should be able to rely on that I think

Reply:
I think this is what we should do

```
func SecretKeystoreFormatMatchesSpec(input Input) (string, string, bool) {
...
_, issuerProvidesCA := input.Secret.Data[cmmeta.TLSCAKey] // check that this issuer provides CA
...
	if input.Certificate.Spec.Keystores.JKS != nil {
		if input.Certificate.Spec.Keystores.JKS.Create {
			if len(input.Secret.Data[cmapi.JKSSecretKey]) == 0 ||
				(len(input.Secret.Data[cmapi.JKSTruststoreKey]) == 0 && issuerProvidesCA) {
				return SecretMismatch, "JKS Keystore keys does not contain data", true
			}
...
```

Reply:
I think I got it know.  :-) Thanks. Will update the code.